build-tests/heft-minimal-rig-usage-test/tsconfig.json

```diff
@@ -1,3 +1,3 @@
 {
-  "extends": "./node_modules/heft-minimal-rig-test/includes/tsconfig-base.json"
+  "extends": "./node_modules/heft-minimal-rig-test/profiles/default/tsconfig-base.json"
```

Comment:
We now support two aspects of rigs:

## 1. dependency resolution

The original Heft "rig package" feature was about [resolving certain packages](https://rushstack.io/pages/heft_tutorials/heft_and_rush/#centralizing-dependencies-in-a-rig-package) based on the `"extends"` field of **tsconfig.json**:  Heft follows the `"extends"` chain to the furthest file, then finds the governing **package.json**, and looks to see if it depends on `typescript`.  If so, then Heft loads the following dependencies from that location instead of from the local project:

-    `typescript`
-    `@microsoft/api-extractor`
-    `eslint`
-    `tslint`

These dependencies also can be omitted for the local project, but for a different reason, that Heft itself depends on them:


-    `webpack` and `webpack-dev-server`
-    `jest` and its core dependencies

(This design could be changed. The algorithm was just something to get `rush-stack-compiler` working with Heft.)

## 2. rig.json for config files

The new config file lookup rule introduced by [rig-package](https://www.npmjs.com/@rushstack/rig-package).

Reply:
So, `heft-minimal-rig-usage-test` originally was a test for feature #<!-- -->1.  If we want it to be a test for feature #<!-- -->2 as well, then I think it should demo every riggable config file supported by Heft.  (We could maintain separate pairs of projects for testing #<!-- -->1 and #<!-- -->2, but I think it's fine to roll them together.)

So let's rename these projects:
- `heft-rig-test` --> `heft-rig-test`
- `heft-rig-usage-test` --> `heft-rig-usage-test`

And let's beef up `heft-rig-test/profile/default` to have an example of every supported file.